libraries/core_libs/network/src/tarcap/packets_handlers/common/ext_votes_packet_handler.cpp

```diff
@@ -19,12 +19,13 @@ ExtVotesPacketHandler::ExtVotesPacketHandler(std::shared_ptr<PeersState> peers_s
       vote_mgr_(std::move(vote_mgr)) {}
 
 std::pair<bool, std::string> ExtVotesPacketHandler::validateStandardVote(const std::shared_ptr<Vote> &vote) const {
-  const uint64_t current_pbft_period = pbft_chain_->getPbftChainSize();
-  const auto current_pbft_round = pbft_mgr_->getPbftRound();
+  const auto [current_pbft_round, current_pbft_period] = pbft_mgr_->getPbftRoundAndPeriod();
 
   // Old vote or vote from too far in the future, can be dropped
   // TODO[1880]: should be vote->getPeriod() <= current_pbft_period - if <=, some tests are failing due to missing
   // reward votes -> whole rewards votes gossiping need to be checked...
+
+  // CONCERN: Why the minus one on the vote period?
```

Comment:
becasue votes are usually for future period (not yet pushed into the chain) so when validating vote in dpos contract, we use vote_period -  1

Reply:
Wouldn't it be more correct to check it on `vote_period - delefation_delay`?